### PR TITLE
Krknwk 19077 usage fault on stress

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: main

  # Do not change after creating PR
  Create_NRF_PR: true
Jenkins:
  test-sdk-sidewalk: master
```

## Description

JIRA ticket: KRKNWK-19077

## Self review

- [x] There is no commented code.
- [x] There are no TODO/FIXME comments without associated issue ticket.
- [x] Commits are properly organized.
- [x] Change has been tested.
- [x] Tests were updated (if applicable).

### CODE REVIEW

File: samples/sid_end_device/src/sidewalk.c [hunk 1]
```diff
@@ -365,11 +365,17 @@ static void state_sidewalk_run(void *o)
 
 static void state_sidewalk_exit(void *o)
 {
-	while (sys_slist_is_empty(&pending_message_list) == false) {
-		sidewalk_msg_t *message;
-		message = SYS_SLIST_CONTAINER(sys_slist_get(&pending_message_list), message, node);
+	//  returns NULL on empty list, or HEAD node.
```
Comment (ktaborowski) [inline samples/sid_end_device/src/sidewalk.c:368]: Those comments aren't much descriptive then the code itself, I think you can remove them

File: samples/sid_end_device/src/sidewalk.c [hunk 2]
```diff
@@ -365,11 +365,17 @@ static void state_sidewalk_run(void *o)
...
+	sys_snode_t *list_element = sys_slist_get(&pending_message_list);
+
+	while (list_element != NULL) {
+		// Get content of node, never NULL, as this macro only apply offset to list_element pointer.
+		sidewalk_msg_t *message =  SYS_SLIST_CONTAINER(list_element, message, node);
 		sid_hal_free(message->msg.data);
```
Comment (ktaborowski) [inline samples/sid_end_device/src/sidewalk.c:374]: Not all messages has data, worth to check if it's not null

File: samples/sid_end_device/src/sidewalk.c (hunk 1 above)
Comment (RobertGalatNordic) [reply samples/sid_end_device/src/sidewalk.c]: removed

**Comment by RobertGalatNordic** [comment]
[13:20] Tyzenhauz, Tomasz
https://jenkins-ncs.nordicsemi.no/job/KRKNG/job/Sidewalk/job/test-sdk-sidewalk/job/test-branch/119/

**Pull request merged**